pandas/tests/arithmetic/test_numeric.py

```diff
@@ -235,6 +235,30 @@ def test_add_sub_timedeltalike_invalid(self, numeric_idx, other, box):
         with pytest.raises(TypeError):
             other - left
 
+    @pytest.mark.parametrize(
+        "other",
+        [
+            pd.Timestamp.now().to_pydatetime(),
+            pd.Timestamp.now(tz="UTC").to_pydatetime(),
+            pd.Timestamp.now().to_datetime64(),
+            pd.NaT,
+        ],
+    )
+    @pytest.mark.filterwarnings("ignore:elementwise comp:DeprecationWarning")
```

Comment:
Who / what is this from?

Reply:
in ensure_python_int we have a 'assert new_value == value` (inside a try/except) that numpy is warning about when `value` is a np.datetime64 obj

Reply:
This raises for me with NumPy 1.17

```python

In [13]: a = np.datetime64(10, 'ns')

In [14]: pd.core.dtypes.common.ensure_python_int(a)
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
~/sandbox/pandas/pandas/core/dtypes/common.py in ensure_python_int(value)
    203         new_value = int(value)
--> 204         assert new_value == value
    205     except (TypeError, ValueError, AssertionError):

AssertionError:

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-14-a45e0cb68b99> in <module>
----> 1 pd.core.dtypes.common.ensure_python_int(a)

~/sandbox/pandas/pandas/core/dtypes/common.py in ensure_python_int(value)
    204         assert new_value == value
    205     except (TypeError, ValueError, AssertionError):
--> 206         raise TypeError(msg.format(type(value), value))
    207     return new_value
    208

TypeError: Wrong type <class 'numpy.datetime64'> for value 1970-01-01T00:00:00.000000010
```

It just warned earlier? If so, then +1

Reply:
> This raises for me with NumPy 1.17

Yah, the assertion raises on master, shouldn't be affected by this PR.  This PR makes `pd.Series([0]) + pd.Timestamp.now().to_datetime64()` raise TypeError, which doesn't happen now.